src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,383 @@
+)
+
+
+@devices.command()
+@device_guid_argument
+@change_device_name_option
+@purge_date_option
+@sdk_options()
+def deactivate(state, device_guid, change_device_name, purge_date):
+    """Deactivate a device within Code42. Requires the device GUID to deactivate."""
+    _deactivate_device(state.sdk, device_guid, change_device_name, purge_date)
+
+
+def _deactivate_device(sdk, device_guid, change_device_name, purge_date):
+    device = sdk.devices.get_by_guid(device_guid)
+    try:
+        sdk.devices.deactivate(device.data["computerId"])
+    except exceptions.Py42BadRequestError:
+        raise Code42CLIError("The device {} is in legal hold.".format(device_guid))
+    except exceptions.Py42NotFoundError:
+        raise Code42CLIError("The device {} was not found.".format(device_guid))
+    except exceptions.Py42ForbiddenError:
+        raise Code42CLIError("Unable to deactivate {}.".format(device_guid))
+    if purge_date or change_device_name:
+        guid = device_guid
+        name = device.data["name"]
+    if purge_date:
+        _update_cold_storage_purge_date(sdk, guid, purge_date)
+    if change_device_name and not name.startswith("deactivated_"):
+        _change_device_name(
+            sdk,
+            device_guid,
+            "deactivated_" + date.today().strftime("%Y-%m-%d") + "_" + name,
+        )
+
+
+def _update_cold_storage_purge_date(sdk, guid, purge_date):
+    archives_response = sdk.archive.get_all_by_device_guid(guid)
+    archive_guid_list = [
+        archive["archiveGuid"]
+        for page in archives_response
+        for archive in page["archives"]
+    ]
+    for archive_guid in archive_guid_list:
+        sdk.archive.update_cold_storage_purge_date(archive_guid, purge_date)
+
+
+def _change_device_name(sdk, guid, name):
+    device_settings = sdk.devices.get_settings(guid)
+    device_settings.name = name
+    sdk.devices.update_settings(device_settings)
+
+
+@devices.command()
+@device_guid_argument
+@format_option
+@sdk_options()
+def show(state, device_guid, format=None):
+    """Print device info. Requires device GUID."""
+    _DEVICE_INFO_KEYS_MAP = OrderedDict()
+    _DEVICE_INFO_KEYS_MAP["name"] = "Name"
+    _DEVICE_INFO_KEYS_MAP["osHostname"] = "Hostname"
+    _DEVICE_INFO_KEYS_MAP["guid"] = "GUID"
+    _DEVICE_INFO_KEYS_MAP["status"] = "Status"
+    _DEVICE_INFO_KEYS_MAP["lastConnected"] = "Last Connected Date"
+    _DEVICE_INFO_KEYS_MAP["productVersion"] = "Code42 Version"
+    _DEVICE_INFO_KEYS_MAP["osName"] = "Operating System"
+    _DEVICE_INFO_KEYS_MAP["osVersion"] = "Operating System Version"
+
+    _BACKUP_SET_KEYS_MAP = OrderedDict()
+    _BACKUP_SET_KEYS_MAP["targetComputerName"] = "Destination"
+    _BACKUP_SET_KEYS_MAP["lastBackup"] = "Last Backup Activity"
+    _BACKUP_SET_KEYS_MAP["lastCompleted"] = "Last Completed Backup"
+    _BACKUP_SET_KEYS_MAP["archiveBytes"] = "Archive Size in Bytes"
+    _BACKUP_SET_KEYS_MAP["archiveGuid"] = "Archive GUID"
```

Comment:
Since we're dropping 3.5, we don't need to use OrderedDict here, since dicts are now ordered by default in 3.6+

And these should maybe be declared outside the function so they can be used elsewhere (tests, etc?) if needed in the future.

Reply:
I originally had them outside the function, and moved them in at @unparalleled-js 's suggestion, since they're not currently used anywhere else, so we don't need to create them unless this particular method is called.

I could move them back out now, but it'd be pretty simple to move them later when/if they end up being re-used. What do you think?

Reply:
I also suggested that we could put them in a helper method.. That way, they can still be used in tests if needed AND that makes them lazily loaded which is better imo

Reply:
moved them to helper methods.